File: app/src/main/java/me/sweetll/evilhide/fragment/SettingFragment.java
package me.sweetll.evilhide.fragment;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import me.sweetll.evilhide.R;
import me.sweetll.evilhide.Settings;

/**
 * Created by sweet on 3/7/16.
 */
public class SettingFragment extends PreferenceFragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.settings);
    }
}

File: app/src/main/java/me/sweetll/evilhide/fragment/SettingFragment.kt
package me.sweetll.evilhide.fragment

import android.content.ComponentName
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.os.Bundle
import android.preference.PreferenceFragment
import me.sweetll.evilhide.MainActivity

import me.sweetll.evilhide.R
import me.sweetll.evilhide.Settings

class SettingFragment : PreferenceFragment(), SharedPreferences.OnSharedPreferenceChangeListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.settings)
    }

    override fun onResume() {
        super.onResume()
        preferenceScreen.sharedPreferences.registerOnSharedPreferenceChangeListener(this)
    }

    override fun onPause() {
        super.onPause()
        preferenceScreen.sharedPreferences.unregisterOnSharedPreferenceChangeListener(this)
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        if (key == Settings.KEY_PREF_INVISIBLE) {
            val selfInvisible = sharedPreferences.getBoolean(key, false)
            val p = activity.packageManager
            val componentName = ComponentName(activity, MainActivity::class.java)
            when (selfInvisible) {
                true -> p.setComponentEnabledSetting(componentName, PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP)
                false -> p.setComponentEnabledSetting(componentName, PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP)
            }
        }
    }
}
